ledgerManager.php

```diff
@@ -0,0 +1,319 @@
+		$this->pdo = new PDO("mysql:host=localhost;dbname=". $database, $username, $password);
+		$this->table = $table;
+	}
+
+	/**
+	 * @param int $transactionId
+	 * @param string $description
+	 * @param float $amount
+	 * @param bool $cleared
+	 * @return bool
+	 */
+	public function updateTransaction($transactionId, $description, $amount, $cleared)
+	{
+		$query = $this->pdo->prepare("
+			update {$this->table}
+			set description = :description,
+			amount = :amount,
+			cleared = :cleared
+			where id = :transactionId
+		");
+		return $query->execute([
+			':description' => $description,
+			':amount' => $amount,
+			':cleared' => $cleared,
+			':transactionId' => $transactionId
+		]);
+	}
+
+	/**
+	 * @param string $description
+	 * @param float $amount
+	 * @param bool $cleared
+	 * @return bool
+	 */
+	public function insertTransaction($description, $amount, $credit)
+	{
+		$query = $this->pdo->prepare("
+			insert into {$this->table}
+			(description, amount, credit)
+			values (:description, :amount, :credit)
+		");
+		return $query->execute([
+			':description' => $description,
+			':amount' => $amount,
+			':credit' => $credit,
+		]);
+	}
+
+	/**
+	 * @param int $transactionId
+	 * @return bool
+	 */
+	public function deleteTransaction($transactionId)
+	{
+		$query = $this->pdo->prepare("
+			delete from {$this->table}
+			where id = :transactionId
+		");
+		return $query->execute([':transactionId' => $transactionId]);
+	}
+
+	/**
+	 * This method returns the sum of all credit or debit transactions
+	 * for (an optional) date range
+	 *
+	 * @param bool $credit
+	 * @param string|null $startDate
+	 * @param string|null $endDate
+	 * @return float
+	 */
+	protected function getTotalSumCreditTransactions($credit = true, $startDate = null, $endDate = null)
+	{
+		$queryString = "
+			select sum(amount) amount
+			from {$this->table}
+			where credit = :credit
+		";
+		if ($startDate) {
+			$queryString .= "and time >= :startDate";
+		}
+		if ($endDate) {
+			$queryString .= "and time <= :endDate";
+		}
+		$query = $this->pdo->prepare($queryString);
+		$query->bindParam(':credit', $credit);
+		if ($startDate) {
+			$query->bindParam(':startDate', $startDate);
+		}
+		if ($endDate) {
+			$query->bindParam(':endDate', $endDate);
+		}
+		$query->execute();
+		$row = $query->fetch(PDO::FETCH_ASSOC);
+		return (float)$row['amount'];
+	}
+
+	/**
+	 * This method returns a balance for a given date range, OR sets the value
+	 * of the current balance to the class property $overallBalance
+	 *
+	 * @param string|null $startDate
+	 * @param string|null $endDate
+	 * @return void|float
+	 */
+	protected function calculateBalance($startDate = null, $endDate = null)
+	{
+		$totalCredits = $this->getTotalSumCreditTransactions(true, $startDate, $endDate);
+		$totalDebits = $this->getTotalSumCreditTransactions(false, $startDate, $endDate);
+		$balance = $totalCredits - $totalDebits;
+		if (!$startDate && !$endDate) {
+			$this->overallBalance = $balance;
```

Comment:
comment that this is the only place that this is being set, for the sole purpose of avoiding the re-calculation of the current running balance...although the overall benefit of caching this value remains to be seen.